pyaedt/emit.py

```diff
@@ -1,16 +1,242 @@
+            )
+        return interaction_domain
+
+    @staticmethod
+    def result_mode_error():
+        """
+        Prints error message.
+
+        Returns
+        -------
+        """
+        print("This function is inaccessible when the Emit object has no revisions")
+
+    @pyaedt_function_handler()
+    def get_radio_names(self, tx_rx):
+        """
+        Return a list of all ``tx'' or ``rx`` radios in the project.
+
+        Parameters
+        ----------
+        tx_rx : tx_rx_mode object
+            Used for determining whether to get ``rx`` or ``tx`` radio names.
+        -------
+        :class:`list of str`
+
+        Examples
+        ----------
+        >>> radios = aedtapp.results.get_radio_names(Emit.tx_rx_mode.rx)
+        """
+        if self.get_result_loaded():
+            radios = self.emit_api.radio_names(tx_rx)
+        else:
+            radios = None
+            Emit.result_mode_error()
+        return radios
+
+    @pyaedt_function_handler()
+    def get_band_names(self, radio_name, tx_rx_mode):
+        """
+        Return a list of all ``tx`` or ``rx`` bands in a given radio.
+
+        Parameters
+        ----------
+        radio_name : str
+            Name of the radio.
+        tx_rx : tx_rx_mode object
+            Used for determining whether to get ``rx`` or ``tx`` radio names.
+        -------
+        :class:`list of str`
+
+        Examples
+        ----------
+        >>> bands = aedtapp.results.get_band_names('Bluetooth', Emit.tx_rx_mode.rx)
+        """
+        if self.get_result_loaded():
+            bands = self.emit_api.band_names(radio_name, tx_rx_mode)
+        else:
+            bands = None
+            Emit.result_mode_error()
+        return bands
+
+    @pyaedt_function_handler()
+    def get_active_frequencies(self, radio_name, band_name, tx_rx_mode):
+
+        """
+        Return a list of active frequencies for a ``tx`` or ``rx`` band in a radio.
+
+        Parameters
+        ----------
+        radio_name : str
+            Name of the radio.
+        band_name : str
+           Name of the band.
+        tx_rx : tx_rx_mode object
+            Used for determining whether to get ``rx`` or ``tx`` radio names.
+        -------
+        :class:`list of float`
+
+        Examples
+        ----------
+        >>> bands = aedtapp.results.get_band_names('Bluetooth', 'Rx - Base Data Rate', Emit.tx_rx_mode.rx)
+        """
+        if self.get_result_loaded():
+            freq = self.emit_api.active_frequencies(radio_name, band_name, tx_rx_mode)
+        else:
+            freq = None
+            Emit.result_mode_error()
+        return freq
+
+
+class Revision:
+    """
+    Provides the ``Revision`` object.
+
+    Parameters
+    ----------
+    Emit_obj :
+         ``Emit`` object that this revision is associated with.
+    name : str, optional
+        Name of the revision to create. The default is ``None``, in which case a
+        default name is given.
```

Comment:
```suggestion
        default name is given.
        
```
A blank line must exist between these headings for doc generation to succeed.